test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -616,6 +616,31 @@ Describe "Hard link and symbolic link tests" -Tags "CI", "RequireAdminOnWindows"
             $ci = Get-ChildItem $alphaLink -Recurse -Name
             $ci.Count | Should -BeExactly 7 # returns 10 - unexpectly recurce in link-alpha\link-Beta. See https://github.com/PowerShell/PowerShell/issues/11614
         }
+        It "Get-ChildItem will recurse into emulated OneDrive directory" -Skip:(-not $IsWindows) {
+            # The test depends on the files created in previous test:
+            #New-Item -ItemType SymbolicLink -Path $alphaLink -Value $alphaDir
+            #New-Item -ItemType SymbolicLink -Path $betaLink -Value $betaDir
+
+            [System.Management.Automation.Internal.InternalTestHooks]::SetTestHook('OneDriveTestOn', $true)
+            [System.Management.Automation.Internal.InternalTestHooks]::SetTestHook('OneDriveTestRecuseOn', $false)
```

Comment:
Rather than this, these tests are probably better within their own context, with a `BeforeAll` and `AfterAll` for the test hooks

Reply:
We could but I again keep the file style and follow a style of previous tests.

Reply:
Ok, well I think it's time to break with this file's style because it's cleaner to do things this way. We've been very happy to write sweeping style changes elsewhere with no conceivable benefit, and in this case there is a clear benefit

Reply:
It will rather add confusion.
Earlier I just made a new FileSystemProviderExtended.Tests.ps1 file when there were many new tests to be created.
If we have to change something, then exactly the same.
We are already aware of many bugs and performance issues in FileSystemProvider. I could fix a lot there including tests. But while my PRs are frozen, I see no reason to change something minor now.